### PR TITLE
antora: replace site-generator-default with site-generator

### PR DESCRIPTION
**Pull request by hqurve**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/155113

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nomeata** [comment]
I was about to test these patches, but they need updating (or maybe you can tell me how to regenerate the large node file?)

**Comment by nomeata** [comment]
This looks useful, if it is the right way to fix #155113. What is the state of this PR?

**Comment by hqurve** [comment]
Last I remember, it works. However, it cant be merged because of the conflict in node-packages.nix. Note because of how node2nix works (or maybe how nixpkgs uses it), this file takes a long time to regenerate and be reviewed.

Personally, I initially switched to just using node2nix to generate a smaller package set (just with antora and other things I need). However, later I switched to just using npm directly where needed to get antora because of node2nix nodejs16 issues.

tldr: it works however I've switched to using a smaller package set/using npm directly

**Comment by nomeata** [comment]
Thanks for the update. Maybe we are switching away from antora anyways, I heard, though, so I’ll let it rest for now.
